### PR TITLE
type hint of all the rest files under `/tests`

### PR DESCRIPTION
**Pull request by gasin**
## Motivation
Please refer to the Issue #711 .

## Description of the changes
I fixed the type annotations of all the rest files under `tests`

### CODE REVIEW

**Review by HideakiImamura — CHANGES_REQUESTED** [review]
Thanks for the exhaustive improvement on the `/tests` directory! I have some comments. Could take a look?

File: tests/samplers_tests/test_samplers.py [hunk 1]
```diff
@@ -1,30 +1,28 @@
 from collections import OrderedDict
 import pickle
+import typing
```
[the remark this answers is not in the document]
Comment (gasin) [reply tests/samplers_tests/test_samplers.py]: Thank you for the comment.
Applied

File: tests/samplers_tests/test_samplers.py [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
 
 import numpy as np
 import pytest
+from typing import Any
+from typing import Dict
+from typing import Optional
+from typing import Sequence
```
[the remark this answers is not in the document]
Comment (gasin) [reply tests/samplers_tests/test_samplers.py]: Sorry for the careless mistake, fixed.

[the remark this answers is not in the document]
Comment (HideakiImamura) [reply tests/samplers_tests/test_samplers.py]: Thanks for the reply! You don't need to feel sorry at all. We are here to find and fix those things 😃

**Review by HideakiImamura — APPROVED** [review]
Thank you for your swift actions! LGTM!

File: tests/samplers_tests/test_samplers.py [hunk 3]
```diff
@@ -289,8 +298,7 @@ def test_intersection_search_space() -> None:
 
     # Failed or pruned trials are not considered in the calculation of
     # an intersection search space.
-    def objective(trial, exception):
-        # type: (optuna.trial.Trial, Exception) -> float
+    def objective(trial: optuna.trial.Trial, exception: Exception) -> float:
```
Comment (hvy) [inline tests/samplers_tests/test_samplers.py:301]: Not introduced by this PR but we can simplify this line.
```suggestion
    def objective(trial: Trial, exception: Exception) -> float:
```

**Review by hvy — CHANGES_REQUESTED** [review]
Thanks for the PR, the changes more or less LGTM. I left one single nit.

File: tests/samplers_tests/test_samplers.py (hunk 3 above)
Comment (gasin) [reply tests/samplers_tests/test_samplers.py]: Thx. Applied

**Review by hvy — APPROVED** [review]
Thanks for your very quick response, LGTM!

Edit: Let me merge this one once CI is happy.

**Pull request merged**